sdk/src/test/scala/com/horizen/account/state/ForgerStakeMsgProcessorTest.scala

```diff
@@ -656,8 +656,100 @@ class ForgerStakeMsgProcessorTest
     }
   }
 
+  @Test
+  def testRejectSendingInvalidValueToWithdraw(): Unit = {
+    val expectedBlockSignerProposition = "aa22334455667788112233445586778811223344556677881122334455667788" // 32 bytes
+    val blockSignerProposition = new PublicKey25519Proposition(BytesUtils.fromHexString(expectedBlockSignerProposition)) // 32 bytes
+    val expectedVrfKey = "aabbccddeeff0099aabb87ddeeff0099aabbccddeeff0099aabbccd2aeff001234"
+    val vrfPublicKey = new VrfPublicKey(BytesUtils.fromHexString(expectedVrfKey)) // 33 bytes
+
+    Mockito.when(mockNetworkParams.restrictForgers).thenReturn(true)
+    Mockito.when(mockNetworkParams.allowedForgersList).thenReturn(Seq(
+      (blockSignerProposition, vrfPublicKey)
+    ))
+
+    usingView(forgerStakeMessageProcessor) { view =>
+
+      forgerStakeMessageProcessor.init(view)
+
+      // create sender account with some fund in it
+      // val initialAmount = BigInteger.valueOf(10).multiply(validWeiAmount)
+      val initialAmount = ZenWeiConverter.MAX_MONEY_IN_WEI
+      createSenderAccount(view, initialAmount)
+
+      val cmdInput = AddNewStakeCmdInput(
+        ForgerPublicKeys(blockSignerProposition, vrfPublicKey),
+        ownerAddressProposition
+      )
+      val addNewStakeData: Array[Byte] = cmdInput.encode()
+
+      val stakeAmount = validWeiAmount
+      val addNewStakeMsg = getMessage(contractAddress, stakeAmount, BytesUtils.fromHexString(AddNewStakeCmd) ++ addNewStakeData, randomNonce)
+      val expStakeId = forgerStakeMessageProcessor.getStakeId(addNewStakeMsg)
+      val forgingStakeInfo = AccountForgingStakeInfo(expStakeId, ForgerStakeData(ForgerPublicKeys(blockSignerProposition, vrfPublicKey), ownerAddressProposition, stakeAmount))
+      val addNewStakeReturnData = withGas(forgerStakeMessageProcessor.process(addNewStakeMsg, view, _, defaultBlockContext))
+      assertNotNull(addNewStakeReturnData)
+
+
+      val nonce = randomNonce
+      val msgToSign = ForgerStakeMsgProcessor.getMessageToSign(forgingStakeInfo.stakeId, origin, nonce.toByteArray)
+      val msgSignatureData = Sign.signMessage(msgToSign, pair, true)
+      val msgSignature = new SignatureSecp256k1(msgSignatureData)
+
+      // create command arguments
+      val removeCmdInput = RemoveStakeCmdInput(forgingStakeInfo.stakeId, msgSignature)
+      val data: Array[Byte] = removeCmdInput.encode()
+      var msg = getMessage(contractAddress, BigInteger.ONE, BytesUtils.fromHexString(RemoveStakeCmd) ++ data, nonce)
+
+      assertThrows[ExecutionFailedException] {
```

Comment:
I think we can be even more precise here with exception type: ExecutionFailedException -> ExecutionRevertedException
Same in all cases below